src/polynomial/univariate.rs

```diff
@@ -7,6 +7,17 @@ use ff::Field;
 /// Coefficients of the polynomial are presented from smaller to larger
 pub struct UnivariatePoly<F: Field>(pub(crate) Box<[F]>);
 
+impl<F: Field> UnivariatePoly<F> {
```

Comment:
where these two functions are used?

Reply:
This is not an abstraction level, but a helper class, since the same struct from itertools is a bit off for us

The only thing I can do is to separate it into a separate utility crate and take it out of sirius and just import it here, but it is very small for such a class.

Reply:
you answered in the wrong comment

Reply:
Yeah, It might be better to add them when accompany with the calling code so that we don't have non-used code.